week3/portfolio_challenge/9_testing_react_with_cypress.md

```diff
@@ -0,0 +1,204 @@
+
+  ## Folder structure
+
+  Cypress scaffolded many folders for you let me walk through which does what:
+
+  - fixtures/
+    - This folder holds your fixture files, mock files in other words. These files can be used to mock network calls.
+    - Some example json files were scaffolded you can remove those we will not use them in this project.
+  - integration/
+    - This folder holds your test files. You can create test files either as `example.spec.js` or as `example_spec.js` cypress will recognize both types. You can create sub-folders to group your test files.
+    - By default an examples folder is scaffolded with a lot of examples by cypress. You can delete those we will not use any of them in this project
+  - plugins/index.js 
+    - This file can be used to load plugins. Plugins enable you to tap into, modify, or extend the internal behavior of Cypress.
+  - support/commands.js
+    - This file allows you to add custom commands to cypress, or overwrite existing ones.
+  - support/index.js
+    - It is processed and loaded automatically before your test files. This is a great place to put global configuration and behavior that modifies Cypress.
+
+In this project we will focus on adding tests in the integration folder.
+
+One more file has been added by the scaffold that is `cypress.json` in the root folder. Here you can configure cypress further. Let's edit this file and add the following code:
+
+```json
+{
+  "baseUrl": "http://localhost:3000"
+}
+```
+
+This will tell cypress that `http://localhost:3000` is the root for the project.
+
+## Creating your first test
+
+Cypress works very similarly to Rspec, it is organized into `describe` and `it` blocks. These are functions that receive 2 arguments, the title of the test and an anonymous function that will execute the next code block. Add the following file:
+
+```js
+// cypress/integration/visitor_can_see_landing_page.spec.js
+
+describe('Visitor can see landing page' , () => {
+  it('visitor can see landing page', () => {
+    cy.visit('/')
+  })
+})
+```
+
+To execute cypress functions you need to use `cy` prefix. The most basic one is `visit()` which will navigate to a url which we add. Since we added the baseUlr as `http://localhost:3000` by typing in `visit('/')` we will navigate just there.
+
+After adding that file, start your development server and run cypress and you should see something like this:
+
+![](./cypress_2.png)
+
+We have our first green test! Now let's see what are the most common cypress commands we can use in our tests files to find elements, interact with them and check their content.
+
+
+## Most common cypress commands 
+
+
+### [cy.get()](https://docs.cypress.io/api/commands/get.html#Syntax)
+- Get one or more DOM elements by selector or alias.
+- Examples:
+  - `cy.get(.some-class)` - gets elements with `some-class` class
+  - `cy.get(#some-id)` - gets elements with `some-id` id
+  - `cy.get(button)` - gets all button elements
```

Comment:
`cy.get('button')` this must be wrapped in quotes, right? Same with the ones above?

Reply:
yes